File: patterns/generic-is-alive/rules-js/generic-is-alive.js
rules.JSRule({
  name: "A sensor stopped reporting",
  triggers: [triggers.GroupStateChangeTrigger("DeviceStatuses", undefined, "UNDEF")],
  execute: (event) => {
    // Meldung oder Alarm ausloesen
    console.warn(event.itemName + " meldet sich nicht mehr (UNDEF) - Alarm/Meldung ausloesen");
  }
});

File: patterns/generic-is-alive/rules-python/generic_is_alive.py
from openhab import rule
from openhab.triggers import GroupStateChangeTrigger


@rule(triggers=[GroupStateChangeTrigger("DeviceStatuses", state="UNDEF")])
class ASensorStoppedReporting:
    def execute(self, module, input):
        event = input.get("event")
        item_name = event.getItemName() if event else "unbekannt"
        self.logger.warn(item_name + " meldet sich nicht mehr (UNDEF) - Alarm/Meldung ausloesen")
        # Meldung oder Alarm ausloesen
